main.go

```diff
@@ -126,11 +130,19 @@ func uploadFile(api *CacophonyAPI, filename string) error {
 	}
 	defer f.Close()
 	br := bufio.NewReader(f)
-	if err := api.UploadThermalRaw(info, br); err != nil {
-		return err
+	for i := 3; i >= 1; i-- {
+		if err := api.UploadThermalRaw(info, br); err != nil {
+			if i >= 2 {
+				log.Printf("upload failed, trying %d more times", i-1)
```

Comment:
If you make `i` zero based instead of one based, then you don't need to subtract `1` here and the code is arguably slightly easier to understand.